workshop-create.py

```diff
@@ -6,89 +6,116 @@
 import json
 import shutil
 import fileinput
-import re 
+import re
 import glob
+import boto3
+import botocore
 
 argparse = argparse.ArgumentParser()
 argparse.add_argument('--dir', help="Workshop directory", required=True)
 args = argparse.parse_args()
+sts = boto3.client('sts')
 
-docker_staging=os.path.join(args.dir, ".docker_staging")
-terraform_staging=os.path.join(args.dir, ".terraform_staging")
+docker_staging = os.path.join(args.dir, ".docker_staging")
+terraform_staging = os.path.join(args.dir, ".terraform_staging")
 
 # Open and parse configuration file
-with open( os.path.join(args.dir, "workshop.yaml"), 'r') as yaml_file:
+with open(os.path.join(args.dir, "workshop.yaml"), 'r') as yaml_file:
     try:
         config = yaml.safe_load(yaml_file)
     except yaml.YAMLError as exc:
         print(exc)
 
-def copytree(src, dst):
-  if not os.path.exists(dst):
-    os.makedirs(dst)
-    shutil.copystat(src, dst)
-  lst = os.listdir(src)
-  for item in lst:
-    s = os.path.join(src, item)
-    d = os.path.join(dst, item)
-    if os.path.isdir(s):
-      copytree(s, d)
+
+def check_login():
+    boto3.setup_default_session(profile_name=(config['workshop']['core']['profile']))
+    sts = boto3.client('sts')
+    try:
+        sts.get_caller_identity()
+        return True
+    except botocore.exceptions.UnauthorizedSSOTokenError:
+        return False
+    except botocore.exceptions.ClientError:
+        return False
+
+
+if (config['workshop']['core']['cloud_provider']) == 'aws':
```

Comment:
Nice solution to the problem. Can we add likewise check to other clouds?

Reply:
yes, working on that for gcp and azure